apps/settings/js/bluetooth.js

```diff
@@ -685,18 +688,25 @@ navigator.mozL10n.once(function bluetoothSettings() {
     }
 
     function setDeviceUnpair(device) {
+      function unpairDevice() {
+        // backend takes responsibility to disconnect first.
+        var req = defaultAdapter.unpair(device.address);
+        req.onerror = function bt_pairError() {
+          showDevicePaired(true, null);
+        };
+      }
+
       if (device.address === connectedAddress) {
-        var msg = _('unpair-title') + '\n' + _('unpair-msg');
-        if (!window.confirm(msg)) {
-          return;
-        }
-        connectedAddress = null;
+        navigator.mozL10n.formatValue('unpair-confirm').then(msg => {
+          if (!window.confirm(msg)) {
+            return;
+          }
```

Comment:
good point. It will be fast, but the idea behind it is that it should have to be to work properly. I'll test it.